src/main/scala/ch/epfl/lamp/CourseraBuild.scala

```diff
@@ -126,35 +123,20 @@ trait CourseraBuild extends sbt.Build {
   val grade = TaskKey[Unit]("grade")
 
   // mapR: submit the grade / feedback in any case, also on failure
-  lazy val gradeSetting = grade <<= (gradingUUID, scalaTestSubmission, styleCheckSubmission, apiKey, gradeProjectDetails, streams) mapR { (uuidR, sts, scs, apiKeyR, projectDetailsR, s) =>
-    val Value(uuid) = uuidR
+  lazy val gradeSetting = grade <<= (scalaTestSubmission, styleCheckSubmission, gradeProjectDetails, streams) mapR { (sts, scs, projectDetailsR, s) =>
     val logOpt = s match {
       case Value(v) => Some(v.log)
       case _ => None
     }
-    logOpt.foreach(_.info(GradingFeedback.feedbackString(uuid, html = false)))
+    val feedback = GradingFeedback.feedbackString(html = false)
+    // Output to logs for instructor debugging purposes.
+    logOpt.foreach(_.info(feedback))
     val Value(projectDetails) = projectDetailsR
-    apiKeyR match {
-      case Value(apiKey) if (!apiKey.isEmpty) =>
-        logOpt.foreach(_.debug("Course Id for submission: " + projectDetails.courseId))
-        logOpt.foreach(_.debug("Corresponding API key: " + apiKey))
-        // if build failed early, we did not even get the api key from the submission queue
-        if (!GradingFeedback.apiState.isEmpty && !Settings.offlineMode) {
-          val scoreString = "%.2f".format(GradingFeedback.totalScore)
-          CourseraHttp.submitGrade(GradingFeedback.feedbackString(uuid), scoreString, GradingFeedback.apiState, apiKey, projectDetails, logOpt) match {
-            case Failure(msgs) =>
-              sys.error(msgs.list.mkString("\n"))
-            case _ =>
-              ()
-          }
-        } else if (Settings.offlineMode) {
-          logOpt.foreach(_.info(" \nSettings.offlineMode enabled, not uploading the feedback"))
```

Comment:
GrID reverses the operations. With the old system, you called Coursera to poll. In the new system, GrID calls you. Therefore, offlineMode doesn't make sense because you're always offline. (In fact, graders in GrID do not have network access for security reasons.)